src/player.cpp

```diff
@@ -6205,6 +6205,12 @@ void player::suffer()
             }
         }
     }
+
+    // Recalculate, as it might have changed (by mod_*_bonus above)
+    str_cur = get_str();
+    int_cur = get_int();
+    dex_cur = get_dex();
+    per_cur = get_per();
```

Comment:
This "fix afterwards" approach doesn't seem quite right. Furthermore, this chunk was copied from `player::process_artifact()` verbatim.

I'd suggest updating stats inside corresponding bonus setters: `set_str_bonus()` updates `str_cur`, `set_int_bonus()` updates `int_cur`, and so on. This would be more robust, enable better encapsulation, and some code deduplication as a bonus.

Reply:
I agree that that would be a much better way to do it. I was trying for the minimal code impact here that would fix the bug. I'll make another PR that has more code changes and the maintainers can choose which fix they'd prefer.

Actually the whole str_cur variable is the non-encaspulated version, Character::get_str() is the correct function to call. It just seems like the refactoring to move away from str_cur is incomplete.

Reply:
#28282 I've solved the bug a different way, only one of these PRs needs to be applied.